hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaSoftwareModuleManagement.java

```diff
@@ -236,7 +242,9 @@ public void delete(final Collection<Long> ids) {
             // delete binary data of artifacts
             deleteGridFsArtifacts(swModule);
 
-            if (isUnassigned(swModule.getId())) {
+            // execute this count operation without access limitations since we have to
+            // ensure it's not assigned when deleting it.
+            if (distributionSetRepository.countByModulesId(swModule.getId()) <= 0) {
```

Comment:
Since this call will be executed against the repository, the access restriction is not applied, correct? Wondering if we should wrapt those kind of executions into an "system context" execution.

Reply:
Assuming that such "countBy" are not covered by auth acm apply they are considered to be executed in system scope. If, at some point, we add such auto for them - they usage shall be reconsidered if to be executed in system scope